TemporaryParser-Tests/TPRBParserTest.class.st

```diff
@@ -168,104 +178,437 @@ TPRBParserTest >> testBlockReturnNode [
+	 node instead. This has some repercutions : their is no cascade message and the name of the parents are wrong."
+	self assert: tree statements first isFaulty.
+	self assert: tree statements first errorMessage equals: 'Message expected'.
+	
+	self assert: tree statements last isFaulty.
+	self assert: tree statements last errorMessage equals: 'End of statement list encountered'.
+	
+	self assert: tree parent printString equals: 'noMethod
+  1; node.
+  1; node'.
+	
+	"The problem is further enlightenned with longer cascades."
+	tree := self parseFaultyExpression: 'faulty cascade; successful; until; not; 1; node'.
+	self assert: tree isSequence.
+	self assert: tree statements size equals: 2.
+	
+	self assert: tree statements first isFaulty.
+	self assert: tree statements first errorMessage equals: 'Message expected'.
+	
+	self assert: tree statements last isFaulty.
+	self assert: tree statements last errorMessage equals: 'End of statement list encountered'.
+	
+	self assert: tree parent printString equals: 'noMethod
+  1; node.
+  1; node'.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageForgottenDotAfterAssignement [
+	| tree child error |
+	self shouldnt: [self parserClass parseExpression: '|temp | temp := 1 + 2 temp + 3'] raise: SyntaxErrorNotification.
+	
+	tree := self parserClass parseExpression: '|temp | temp := 1 + 2 temp + 3'.
+	self assert: tree isSequence.
+	
+	"There should only be one statement as there is no dot."
+	child := tree statements.
+	self assert: (child size) equals: 1.
+	self assert: child first isAssignment.
+	
+	"Getting the part changed by missing dot."
+	error := child first value receiver arguments first.
+	self assert: error isMessage.
+	self assert: error receiver value equals: 2.
+	self assert: error selector equals: #temp
+	
+	
+	
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageForgottenReceiver [
+	"Test made in sight of implementing the search of previous litterals when encountering an unknown 
+	 variable in case it is, in fact, a selector. This works for cascade (forgotten ;) or missing 
+	 receiver like self ..."
+	self should: [self parserClass parseExpression: '|temp | temp := 1 + 2. + 3'] raise: SyntaxErrorNotification.
+]
+
+{ #category : #testIntervals }
+TPRBParserTest >> testGeneralBlockIntervalsArray [
+	"This tests that array nodes inside a block node correspond the right interval in the source code."
+	| tree str |
+	tree := (self parseExpression: '["comment node" | temporaries | message node. #literalNode . #(literal array node). {array node}. [block node]. cascade node; cascaded. assignement := node. ^return node]').
+	str := '{array node}'.
+	self assert: ((tree body statements at: 4) start) equals: (tree source findString: str).
+	self assert: ((tree body statements at: 4) stop) equals: ((tree source findString: str) + (str size - 1)).

```

Comment:
using `findString:` is nice, it avoids hardcoding the thing.

I'd also go a bit further and refactor it to make it more readable:
```
arrayNode := tree body statements at: 4.
position := tree source findString: str.
self assert: arrayNode start equals: position.
self assert: arrayNode stop equals: position + (str size - 1).
```